[ERROR] Error while decoding to JSON: Malformed UTF-8 characters, possibly incorrectly encoded[99] in /var/www/vendor/phper666/jwt-auth/src/JWT.php

# `[ERROR] Error while decoding to JSON: Malformed UTF-8 characters, possibly incorrectly encoded[99] in /var/www/vendor/phper666/jwt-auth/src/JWT.php`

- `Authorization` 值如果随便赋值,导致数据无法解析, 会直接报系统级错误, 不是很友好

这周末优化一下





---Original---
From: "dlf8090"<notifications@github.com&gt;
Date: Wed, Sep 2, 2020 10:03 AM
To: "phper666/jwt-auth"<jwt-auth@noreply.github.com&gt;;
Cc: "Subscribed"<subscribed@noreply.github.com&gt;;
Subject: [phper666/jwt-auth] [ERROR] Error while decoding to JSON: Malformed UTF-8 characters, possibly incorrectly encoded[99] in /var/www/vendor/phper666/jwt-auth/src/JWT.php (#32)




 
[ERROR] Error while decoding to JSON: Malformed UTF-8 characters, possibly incorrectly encoded[99] in /var/www/vendor/phper666/jwt-auth/src/JWT.php
  
Authorization 值如果随便赋值,导致数据无法解析, 会直接报系统级错误, 不是很友好 
 
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.

--- a/src/JWT.php
+++ b/src/JWT.php
@@ -443,8 +443,11 @@ class JWT extends AbstractJWT
         if($token == null) {
             $token = JWTUtil::getToken($this->request);
         }
-
-        return JWTUtil::getParser()->parse($token);
+        try{
+            return JWTUtil::getParser()->parse($token);
+        }catch (\Exception $e) {
+            throw new JWTException('Jwt token interpretation error. Please provide the correct jwt token and parse the error information: ' . $e->getMessage(), 400);
+        }
     }
 
     public function setScene(string $scene = 'default'): JWT

